__tests__/app.test.js

```diff
@@ -175,6 +249,42 @@ describe("Error Handling", () => {
         expect(body.msg).toBe("Invalid URL");
       });
   });
+  describe("Query Error Handling", () => {
+    it("GET- 400: Should respond with invalid query when passed an invalid column name in sort_by", () => {
+      return request(app)
+        .get("/api/articles?sort_by=whatcolumn")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("sort_by 'whatcolumn' is not allowed.");
+        });
+    });
+    it("GET- 400: Should respond with 400 when given an incorrect topic", () => {
+      return request(app)
+        .get("/api/articles?topic=coding")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("topic 'coding' does not exist.");
+        });
+    });
+    it("GET- 400: Should respond with 400 when given an invalid order by value", () => {
+      return request(app)
+        .get("/api/articles?order=down")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe(
+            "order 'down' does not exist. Please use: 'latest' or 'oldest'"
+          );
+        });
+    });
+    it("GET- 400: Should respond with 400 when all but one query is valid", () => {
+      return request(app)
+        .get("/api/articles?order=oldest&sort_by=whatcolumn&topic=mitch")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("sort_by 'whatcolumn' is not allowed.");
+        });
+    });
```

Comment:
You can add another test here for your greenlight list - check if order, sort_by, or topic is misspelled.